Rewrote Js module documentation

glennsl opened the pull request:
* Added some copyright headers
* Renamed typed_array (+ tests) to js_typed_array
* Added TypedArray as submodule to Js module

Tags don't seem to work with includes anymore, so I commented those out and wrote a couple new sections. Also tried to classify the submodules a bit, and I'm sure you'll have some opinions about that :)

bobzhang commented on docs/blog/index.html:
```diff
@@ -4,7 +4,7 @@
 <meta name="viewport" content="width=device-width, initial-scale=1.0">
-<meta name="generator" content="Asciidoctor 1.5.4">
+<meta name="generator" content="Asciidoctor 1.5.5">
 <title>Bloomberg announcing BuckleScript 1.0</title>
```
can you not check in generated html? it may result in some inconsistency if we have different versions of asciidoctor

glennsl commented:
I've rebased and removed the generated asciidoc html.

Do you mean having separate ocamldoc-generated API docs? I've never tried ocamldoc, but I suppose I agree :)

bobzhang commented:
@glennsl I would prefer just document those 3 modules in the manual, but create ocamldoc html pages for the rest and provide a link, what do you think? (The rational is that only those 3 modules are the core of bucklescript and stability is the key)

glennsl commented:
The 3 modules being `Null`, `Undefined` and `Null_undefined`? Isn't `Boolean` also somewhat core? (Even if it's just the one function that can't be included in `Js`)

I'll have to look into ocamldoc and how to set it up within the project, but in the meantime I think it's a good idea to include these source code links at least.

bobzhang commented:
@glennsl We already have `Js.true_` `Js.false_`, plus in the FFI we are going to have turn `Js.boolean` into `bool` automatically, so it might not be as important as the other 3 modules

glennsl commented:
@bobzhang I updated the links in the Manual to point to the new API reference. Is there anything more you'd like me to do here?

bobzhang commented:
hi @glennsl thanks! would you not check in changes to html files? It is hard to review changes. I have a watch mode which will rebuild html files whenever I checkout

glennsl commented:
@bobzhang Ok, removed the generated html
